src/components/TodoList/TodoList.js

```diff
@@ -1,41 +1,69 @@
 import React from 'react';
 import './TodoList.css';
 
-function handleClick() {
-  console.log(this)
-}
-
 class TodoList extends React.Component {
 
-  // state = {
-  //   list: [
-  //     { id: 1, name: 'Fumer des cannes', active: true },
-  //     { id: 2, name: 'Coder cette shit', active: true },
-  //     { id: 3, name: 'Nager avec des baleines', active: true },
-  //     { id: 4, name: 'Rouler des cannes pour demain', active: true }
-  //   ],
-  //   newTaskName: "",
-  // // }
-  //
-  // handleClick = () => {
-  //   console.log(this)
-  // }
+  state = {
+    list: [ ],
+    newTaskName: "",
+    editTaskName: ""
+  }
 
   deleteTodo(itemToDeleteId) {
     const newList = this.state.list.filter((item) => item.id !== itemToDeleteId)
     this.setState({ list: newList })
   }
 
   createNewTask() {
+    if (this.state.list.length === 0) {
+      this.setState({
+        list: [{ id: 1, name: this.state.newTaskName, active: true, editing: false }],
+        newTaskName: ""
+      })
+      return;
+    }
+
     this.setState({
       list: [
         ...this.state.list,
-        { id: this.state.list[this.state.list.length - 1].id + 1, name: this.state.newTaskName, active: true }
+        { id: this.state.list[this.state.list.length - 1].id + 1, name: this.state.newTaskName, active: true, editing: false}
       ],
-      newTaskName: ""
+      newTaskName: "",
+      editTaskName: ""
     })
   }
 
+  displayEdit(taskId) {
+    const { list } = this.state;
+    const taskIndex = list.findIndex(task => task.id == taskId)
+    const task = list[taskIndex]
+
+    this.setState({ list: [
+      ...list.slice(0, taskIndex),
+    {
+      id: task.id,
+      name: task.name,
+      active: task.active,
+      editing: !task.editing
```

Comment:
With es6, you can use destructuring on objects too, like we did with the array, using `...`. Type "es6 destructuring" on Google if you want to see more examples and how it works.
Here for instance you can do: 
```javascript
{ 
  ...task, 
  editing: !task.editing 
}
```
`...task` will copy every property from task to the new object you create.